rescan.go

```diff
@@ -482,24 +444,28 @@ func rescan(chain ChainSource, options ...RescanOption) error {
 	//
 	// TODO(wilmer): refactor this and handleBlockDisconnected into their
 	// own methods.
-	handleBlockConnected := func(ntfn *blockntfns.Connected) error {
+	handleBlockConnected := func(ntfn *blockntfns.Connected, isRetry bool) error {
+		// If we have any blocks to retry and the notification doesn't
+		// correspond to the current block being retried, we'll queue it
+		// up to retry after.
+		newHeader := ntfn.Header()
+		if len(retryBlocks) > 0 {
+			retryHeader := retryBlocks[0].Header()
+			if retryHeader.BlockHash() != newHeader.BlockHash() {
+				log.Debug("Stashing", ntfn)
+				retryBlocks = append(retryBlocks, ntfn)
+				return nil
+			}
+		}
+
```

Comment:
Shouldn't we also return if we're getting a notification to try a block that we're already attempting to retry?

Reply:
The assumption is that wouldn't happen given the correctness of the notification source, i.e. duplicate notifications wouldn't be dispatched for the same block. If the notification comes through with `isRetry=false` and it's already being retried, we'd mark ourselves as not current and properly catch up.